### DISCUSSION TITLE
add a Name field in the CodeNotes table

### DISCUSSION THREAD

**Discussion by meleu** [RetroAchievements/RAWeb#783] [Ideas]
It would be **extremely** useful and user-friendlier if achievement creators could add a label to a "CodeNote" and then use such label when coding their achievements (where they currently use only addresses).

As we're starting to support more "modern" consoles, the achievements are becoming more and more complex, with several requirements referring to several different addresses... The cognitive load to properly code an achievement and keep it bug-free is quite huge!

Actually this feature on RAWeb side would be simple to implement. A more careful work would be needed on RAIntegration's side (convert names into addresses and what to do if there's no address with the given name). Then I would like to know @Jamiras opinion here.

---

 **[off-topic comment]**

Something for v2: many names in the DB should be renamed to something more meaningful. Like in these examples:
- CodeNotes -> AddressNotes or MemoryNotes
- MemAddr ->AchievementCode

**Comment by Jamiras** [comment]
As long as the MemAddr still contains the address, the translation to/from aliases would only occur within the achievement editor, and validation could be done at that point. This allows renaming aliases without breaking anything, and the largest amount of backwards compatibility.

A similar idea is https://github.com/RetroAchievements/RAIntegration/issues/478, but that takes the opposite approach. A variable could be defined as a constant address and achieve the same functionality. However, I believe using aliases for code note addresses would be much simpler, both for the user, and for the runtime.

I'm more concerned about having unique aliases than undefined ones. If someone gives the same label to two addresses, its easy enough to map the address to the label, but if the label matches multiple addresses, which one do we use?

**Comment by meleu** [comment]
> I'm more concerned about having unique aliases than undefined ones.

Fair point. We can handle this on the backend (do not allow duplicated aliases for the same GameID in the CodeNotes table). Probably it can also be enforced on the Achievement Editor.
